types/three/examples/jsm/exporters/GLTFExporter.d.ts

```diff
@@ -1,20 +1,126 @@
+
+    /**
+     * Export just the attributes within the drawRange, if defined, instead of exporting the whole array. Default is true.
+     */
     truncateDrawRange?: boolean;
+
+    /**
+     * Export in binary (.glb) format, returning an ArrayBuffer. Default is false.
+     */
+    binary?: boolean;
+
+    /**
+     * Export with images embedded into the glTF asset. Default is true.
+     */
     embedImages?: boolean;
+
+    /**
+     * Restricts the image maximum size (both width and height) to the given value. This option works only if embedImages is true. Default is Infinity.
+     */
+    maxTextureSize?: number;
+
+    /**
+     * List of animations to be included in the export.
+     */
     animations?: AnimationClip[];
+
+    /**
+     * Generate indices for non-index geometry and export with them. Default is false.
+     */
     forceIndices?: boolean;
-    forcePowerOfTwoTextures?: boolean;
+
+    /**
+     * Export custom glTF extensions defined on an object's userData.gltfExtensions property. Default is false.
+     */
     includeCustomExtensions?: boolean;
 }
 
 export class GLTFExporter {
     constructor();
 
-    parse(input: Object3D, onCompleted: (gltf: object) => void, options: GLTFExporterOptions): void;
-    parseAsync(input: Object3D, options: GLTFExporterOptions): Promise<void>;
+    register(callback: (writer: GLTFWriter) => GLTFExporterPlugin): this;
+    unregister(callback: (writer: GLTFWriter) => GLTFExporterPlugin): this;
+
+    /**
+     * Generates a .gltf (JSON) or .glb (binary) output from the input (Scenes or Objects)
+     *
+     * @param input Scenes or objects to export. Valid options:
+     * - Export scenes
+     *   ```js
+     *   exporter.parse( scene1, ... )
+     *   exporter.parse( [ scene1, scene2 ], ... )
+     *   ```
+     * - Export objects (It will create a new Scene to hold all the objects)
+     *   ```js
+     *   exporter.parse( object1, ... )
+     *   exporter.parse( [ object1, object2 ], ... )
+     *   ```
+     * - Mix scenes and objects (It will export the scenes as usual but it will create a new scene to hold all the single objects).
+     *   ```js
+     *   exporter.parse( [ scene1, object1, object2, scene2 ], ... )
+     *   ```
+     * @param onDone Will be called when the export completes. The argument will be the generated glTF JSON or binary ArrayBuffer.
+     * @param onError Will be called if there are any errors during the gltf generation.
+     * @param options Export options
+     */
+    parse(
+        input: Object3D | Object3D[],
+        onDone: (gltf: ArrayBuffer | { [key: string]: any }) => void,
+        onError: (error: ErrorEvent) => void,
+        options?: GLTFExporterOptions,
+    ): void;
+
+    /**
+     * @deprecated parse() expects options as the fourth argument now.
+     */
+    parse(
+        input: Object3D | Object3D[],
+        onDone: (gltf: { [key: string]: any }) => void,
+        options?: GLTFExporterOptions,
+    ): void;
```

Comment:
how do you think about this `@deprecated` signature?

Reply:
I would get rid of this, because if someone used the method like this it would not work.

Reply:
It does work since the compat is still implemented: https://github.com/mrdoob/three.js/blob/c7d06c02e302ab9c20fe8b33eade4b61c6712654/examples/jsm/exporters/GLTFExporter.js#L93-L122

I assume that your potential opinion alternatives can be:

- "Using deprecated functions must be punished from TypeScript type checking".
- "We just don't want to track deprecated functions in the repository".

I will agree in either way.
Just want to make sure your / the repository's opinion for my further contributions 😄

Reply:
> "We just don't want to track deprecated functions in the repository".
I think this is the most straight forward for us.

Reply:
done ✅